src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -314,6 +314,33 @@ void Terminal::ToggleMarkMode()
     }
 }
 
+// Method Description:
+// - switch the targeted selection endpoint with the other one (i.e. start <--> end)
+void Terminal::SwitchSelectionEndpoint()
+{
+    if (IsSelectionActive())
+    {
+        if (WI_IsFlagSet(_selectionEndpoint, SelectionEndpoint::Start) && WI_IsFlagSet(_selectionEndpoint, SelectionEndpoint::End))

```

Comment:
You can use `WI_AreAllFlagsSet` here.

Reply:
For some random reason, it fails to find the `&` operator. Here's the build log:
<details>
<summary> build log </summary>

```
10>D:\Projects\terminal\packages\Microsoft.Windows.ImplementationLibrary.1.0.220201.1\include\wil\Common.h(713,1): error C2678: binary '&': no operator found which takes a left-hand operand of type 'TVal' (or there is no acceptable conversion)
10>        with
10>        [
10>            TVal=Microsoft::Terminal::Core::Terminal::SelectionEndpoint
10>        ] (compiling source file ..\TerminalSelection.cpp)
10>C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.32.31114\include\xfilesystem_abi.h(228,1): message : could be '__std_fs_copy_options operator &(__std_fs_copy_options,__std_fs_copy_options) noexcept' (compiling source file ..\TerminalSelection.cpp)
10>C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.32.31114\include\xfilesystem_abi.h(203,1): message : or       '__std_fs_file_flags operator &(__std_fs_file_flags,__std_fs_file_flags) noexcept' (compiling source file ..\TerminalSelection.cpp)
10>C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.32.31114\include\xfilesystem_abi.h(195,1): message : or       '__std_access_rights operator &(__std_access_rights,__std_access_rights) noexcept' (compiling source file ..\TerminalSelection.cpp)
10>C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.32.31114\include\xfilesystem_abi.h(121,1): message : or       '__std_fs_stats_flags operator &(__std_fs_stats_flags,__std_fs_stats_flags) noexcept' (compiling source file ..\TerminalSelection.cpp)
10>C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.32.31114\include\xfilesystem_abi.h(78,1): message : or       '__std_fs_file_attr operator &(__std_fs_file_attr,__std_fs_file_attr) noexcept' (compiling source file ..\TerminalSelection.cpp)
10>D:\Projects\terminal\packages\Microsoft.Windows.ImplementationLibrary.1.0.220201.1\include\wil\Common.h(713,1): message : while trying to match the argument list '(TVal, TFlags)'
10>        with
10>        [
10>            TVal=Microsoft::Terminal::Core::Terminal::SelectionEndpoint
10>        ]
10>        and
10>        [
10>            TFlags=Microsoft::Terminal::Core::Terminal::SelectionEndpoint
10>        ] (compiling source file ..\TerminalSelection.cpp)
10>D:\Projects\terminal\src\cascadia\TerminalCore\TerminalSelection.cpp(324): message : see reference to function template instantiation 'bool wil::details::AreAllFlagsSetHelper<Microsoft::Terminal::Core::Terminal::SelectionEndpoint,Microsoft::Terminal::Core::Terminal::SelectionEndpoint>(TVal,TFlags)' being compiled
10>        with
10>        [
10>            TVal=Microsoft::Terminal::Core::Terminal::SelectionEndpoint,
10>            TFlags=Microsoft::Terminal::Core::Terminal::SelectionEndpoint
10>        ]
10>D:\Projects\terminal\packages\Microsoft.Windows.ImplementationLibrary.1.0.220201.1\include\wil\Common.h(713,1): fatal error C1001: Internal compiler error.
```

</details>

🤷 we looked at it together for a minute and couldn't figure it out. If anybody comes along and has any ideas, let me know hahaha.